File: qasdk/src/main/java/co/astrnt/qasdk/dao/SummaryQuestionDeserializer.java
package co.astrnt.qasdk.dao;

import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonParseException;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

public class SummaryQuestionDeserializer implements JsonDeserializer<SummaryQuestionApiDao> {

    @Override
    public SummaryQuestionApiDao deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {

        JsonObject object = json.getAsJsonObject();

        if (object.get("question_id").isJsonArray()) {

            List<Integer> ids = new ArrayList<>();

            for (JsonElement item : object.get("question_id").getAsJsonArray()) {
                ids.add(item.getAsInt());
            }

            List<Integer> answerStatus = new ArrayList<>();
            for (JsonElement item : object.get("answered_counter").getAsJsonArray()) {
                answerStatus.add(item.getAsInt());
            }

            return new SummaryQuestionApiDao(ids, answerStatus);
        } else {
            int id = object.get("question_id").getAsInt();
            int answerStatus = object.get("answered_counter").getAsInt();

            return new SummaryQuestionApiDao(id, answerStatus);
        }
    }
}

File: qasdk/src/main/java/co/astrnt/qasdk/dao/SummaryQuestionDeserializer.kt
package co.astrnt.qasdk.dao

import com.google.gson.JsonDeserializationContext
import com.google.gson.JsonDeserializer
import com.google.gson.JsonElement
import com.google.gson.JsonParseException
import java.lang.reflect.Type
import java.util.*

class SummaryQuestionDeserializer : JsonDeserializer<SummaryQuestionApiDao> {
    @Throws(JsonParseException::class)
    override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext): SummaryQuestionApiDao {
        val `object` = json.asJsonObject
        return if (`object`["question_id"].isJsonArray) {
            val ids: MutableList<Int> = ArrayList()
            for (item in `object`["question_id"].asJsonArray) {
                ids.add(item.asInt)
            }
            val answerStatus: MutableList<Int> = ArrayList()
            for (item in `object`["answered_counter"].asJsonArray) {
                answerStatus.add(item.asInt)
            }
            SummaryQuestionApiDao(ids, answerStatus)
        } else {
            val id = `object`["question_id"].asInt
            val answerStatus = `object`["answered_counter"].asInt
            SummaryQuestionApiDao(id, answerStatus)
        }
    }
}